client/client.go

```diff
@@ -179,13 +179,13 @@ func CheckRedirect(_ *http.Request, via []*http.Request) error {
 //		client.FromEnv,
 //		client.WithAPIVersionNegotiation(),
 //	)
-func NewClientWithOpts(ops ...Opt) (*Client, error) {
+func NewClientWithOpts(ctx context.Context, ops ...Opt) (*Client, error) {
 	hostURL, err := ParseHostURL(DefaultDockerHost)
```

Comment:
This will likely be the tricky bit to look at; should a client get a context, or should the context be scoped to an action (like `client.MakeSomeAPIRequest`). I'm included to say for the client, it would be the latter (per call), and it may be up to the consumer of the client to keep a "global" context if things must be inherited from that. 🤔

Reply:
I think this is a similar argument of otel trace visibility over actual context usage (like cancelling io bound tasks).